pymedphys/tests/experimental/pseudonymisation/test_pseudonymisation.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import os
+import subprocess
+import tempfile
+from os.path import exists
+from os.path import join as pjoin
+from shutil import copyfile
+
+import pytest
+
+import pydicom
+
+from pymedphys._dicom.anonymise import (
+    anonymise_file,
+    is_anonymised_directory,
+    is_anonymised_file,
+)
+from pymedphys._dicom.constants.core import DICOM_SOP_CLASS_NAMES_MODE_PREFIXES
+from pymedphys._dicom.utilities import remove_file
+from pymedphys._experimental.pseudonymisation import (
+    get_default_pseudonymisation_keywords,
+    strategy,
+)
+from pymedphys.tests.dicom.test_anonymise import get_test_filepaths
+
+
+@pytest.mark.pydicom
+def test_pseudonymise_file():
+    for test_file_path in get_test_filepaths():
+        _test_pseudonymise_file_at_path(test_file_path)
+
+
+def _test_pseudonymise_file_at_path(test_file_path):
+    assert not is_anonymised_file(test_file_path)
+    identifying_keywords_for_pseudo = get_default_pseudonymisation_keywords()
+    logging.info("Using pseudonymisation keywords")
+    replacement_strategy = strategy.pseudonymisation_dispatch
+    logging.info("Using pseudonymisation strategy")
+
+    with tempfile.TemporaryDirectory() as output_directory:
+        pseudonymised_file_path = anonymise_file(
+            dicom_filepath=test_file_path,
+            output_filepath=output_directory,
+            delete_original_file=False,
+            anonymise_filename=True,
+            replace_values=True,
+            # keywords_to_leave_unchanged=None,
+            delete_private_tags=True,
+            delete_unknown_tags=True,
+            replacement_strategy=replacement_strategy,
+            identifying_keywords=identifying_keywords_for_pseudo,
+        )
+        # debug print + Assert to force the print
+        # print("Pseudonymised file at: ", pseudonymised_file_path)
+        # assert False
+        assert exists(pseudonymised_file_path)
+        ds_input = pydicom.dcmread(test_file_path, force=True)
+        ds_pseudo = pydicom.dcmread(pseudonymised_file_path, force=True)
+        # simplistic stand-in to make sure *something* is happening
+        assert ds_input["PatientID"].value != ds_pseudo["PatientID"].value
+        # make sure that we are not accidentally using the hardcode replacement approach
+        assert ds_pseudo["PatientID"].value not in ["", "Anonymous"]
+
+
+@pytest.mark.slow
+@pytest.mark.pydicom
+@pytest.mark.skipif(
+    "SUBPACKAGE" in os.environ, reason="Need to extract CLI out of subpackages"
+)
```

Comment:
Yeah, I'm realising that what is making this hard to review, it is hard to know which bits have been changed. Copying the file into experimental and making changes, although it's okay, given the fact it's in experimental, it is hard to review.

Reply:
The only bits that were really copied was the invocation of the methods that are in the "stable" (non-experimental) package.
In essence, the strategy and keywords are what has changed (they are new).